tools/test/iree-benchmark-executable.mlir

```diff
@@ -0,0 +1,85 @@
+// Tests the iree-benchmark-executable tool against the portable VMVX target.
+// Other backends can be tested by using the appropriate compiler flags and
+// matching device and executable format flags.
+//
+// Examples:
+//   --iree-hal-target-backends=vmvx
+//     --device=local-sync or --device=local-task
+//     --executable_format=vmvx-bytecode-fb
+//   --iree-hal-target-backends=llvm-cpu
+//     --device=local-sync or --device=local-task
+//     --executable_format=embedded-elf-x86_64
+//     --executable_format=system-dll-x86_64
+//   --iree-hal-target-backends=vulkan-spirv
+//     --device=vulkan
+//     --executable_format=vulkan-spirv-fb
+
+// RUN: iree-compile \
+// RUN:     --compile-mode=hal-executable \
+// RUN:     --iree-hal-target-backends=vmvx \
+// RUN:     %s | \
+// RUN: iree-benchmark-executable \
+// RUN:     --device=local-sync \
+// RUN:     --executable_format=vmvx-bytecode-fb \
+// RUN:     --executable_file=- \
+// RUN:     --entry_point=0 \
+// RUN:     --binding=512xf32 \
+// RUN:     --binding=512xf32 \
+// RUN:     --binding=512xf32 \
+// RUN:     --workgroup_count=1,1,1 \
+// RUN:     --workgroup_count=512,1,1 | \
+// RUN: FileCheck %s --check-prefix=CHECK-VMVX
```

Comment:
I originally had other targets but reduced it to vmvx because we removed the disable checks for optional ones. Would be nice to have one per target somewhere but I'm not sure where that would live now. @ScottTodd?

Reply:
Ah true. https://github.com/openxla/iree/pull/16225 removed those because they were doing tricky things with filtering and CI builds with/without different backends.

For this sort of test that needs to use lit, my first thought is new test suites in `tools/test/` (or `tests/e2e/tools/`...?) That check the `IREE_HAL_DRIVER_` options, like this one: https://github.com/openxla/iree/blob/0d5bf27cb30cfa7f9a71bd7ff2818c8727c5f13e/tools/test/CMakeLists.txt#L57-L62

Basically these are okay:
```cmake
if(CONDITIONS FOR ONE BACKEND)
  tests_for_one_backend()
endif()

if(CONDITIONS FOR ANOTHER BACKEND)
  tests_for_one_backend()
endif()

tests_for_one_backend(
  ...
  LABELS
    "driver=vulkan"  
)
```

but these are not:

```cmake
if (CONDITIONS FOR MULTIPLE BACKENDS)
  tests_for_multiple_backends()
endif()

# this is what we had before
tests_for_multiple_backends(
  ...
  LABELS
    "driver=local-task"
    "driver=metal"
    "driver=vulkan"
)
```